blackboxopt/base.py

```diff
@@ -37,6 +38,17 @@ class ContextError(ValueError):
     """Raised on incomplete or missing context information."""
 
 
+class EvaluationsError(ValueError):
+    """Raised on invalid evaluations (e.g. missing evaluation specificiation id).
+
+    The problematic evaluations are passed in the `evaluations` attribute.
+    """
+
+    def __init__(self, message, evaluations):
+        self.message = message
+        self.evaluations = evaluations
```

Comment:
Let's add type hints here
```python
    def __init__(self, message: str, evaluations: List[Evaluation]):
        self.message = message
        self.evaluations = evaluations
```

I'm also wondering if, for more complex error handling, it might pay to return more specific error details with each evaluation (think something is wrong with the objectives in one evaluation and another one is missing an optimizer specific ID).

Reply:
Type hints: :+1: 
More complex information: I wouldn't do that in this PR. If you see a requirement for that, we should think about a solution which we can also apply to our other custom exceptions. What do you think?

Reply:
See [here](https://github.com/boschresearch/blackboxopt/pull/19/files#r704078451) for the requirement.